test/unit/L1/l1LPTGateway.test.ts

```diff
@@ -42,18 +70,379 @@ describe('L1 Gateway', function() {
     );
 
     l1Gateway = await L1Gateway.deploy(
-        l1GatewayRouter,
-        L2_COUNTERPART,
+        mockL1RouterEOA.address,
+        mockL2GatewayEOA.address,
         escrow.address,
         token.address,
-        L2_LPT,
-        inbox,
+        mockL2LptEOA.address,
+        mockInboxEOA.address,
     );
     await l1Gateway.deployed();
+
+    await token.transfer(sender.address, initialTotalL1Supply);
+    await token.connect(sender).approve(l1Gateway.address, depositAmount);
+
+    const GOVERNOR_ROLE = ethers.utils.solidityKeccak256(
+        ['string'],
+        ['GOVERNOR_ROLE'],
+    );
+    await l1Gateway.grantRole(GOVERNOR_ROLE, governor.address);
+
+    inboxMock = await smock.fake(InboxABI.abi, {
```

Comment:
Since we're already checking in the Inbox interface we can actually avoid also checking in the ABI by using smock to fake the contract using the interface. See [this example](https://github.com/livepeer/arbitrum-lpt-bridge/blob/df3f98c39f52ed70226f120a5156ff435eeaa2bb/test/unit/L1/l1Migrator.test.ts#L117).

Reply:
resolved in #18